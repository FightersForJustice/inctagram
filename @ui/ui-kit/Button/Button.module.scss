@import '../../design/settings/colors.scss';

.button {
  position: relative;
  display: inline-block;
  box-sizing: border-box;
  min-width: 180px;
  height: 36px;
  padding: 6px 24px;
  margin: 5px;

  font-weight: 600;
  font-size: 16px;
  text-align: center;
  line-height: 24px;
  color: $light-100;

  background-color: $primary-500;
  border: none;
  border-radius: 2px;

  cursor: pointer;

  transition: box-shadow 0.2s ease-out;

  &Primary {
    background-color: $primary-500;

      &:hover {
        background-color: $primary-100;
      }

      &:active {
        background-color: $primary-700;
      }

      &:focus {
        background-color: $primary-500;
        outline: 2px solid $primary-700;
      }

      &:disabled {
        cursor: default;
        background-color: $primary-900;
        color: $light-900;
      }
  }

  &Basic {
    background-color: $dark-300;

      &:hover {
        background-color: $dark-100;
      }

      &:active {
        background-color: $dark-400;
      }

      &:focus {
        background-color: $dark-300;
        outline: 1px solid $primary-300;
      }

      &:disabled {
        cursor: default;
        background-color: $dark-500;
        color: $light-900;
      }
  }

  &Outlined {
    background-color: transparent;
    outline: 1px solid $primary-500;
    color: $primary-500;

      &:hover {
        outline-color: $primary-100;
        color: $primary-100;
      }

      &:active {
        outline-color: $primary-700;
        color: $primary-700;
      }

      &:focus {
        outline: 2px solid $primary-700;
        color: $primary-700;
      }

      &:disabled {
        cursor: default;
        outline-color: $primary-900;
        color: $primary-900;
      }
  }

  &Ghost {
    background-color: transparent;
    min-width: 100px;
    color: $primary-500;

      &:hover {
        color: $primary-100;
      }

      &:active {
        color: $primary-700;
      }

      &:focus {
        outline: 2px solid $primary-700;
        color: $primary-500;
      }

      &:disabled {
        cursor: default;
        color: $primary-900;
      }
  }
}
