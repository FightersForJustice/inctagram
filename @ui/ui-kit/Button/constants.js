export const BUTTON_COLORS = Object.freeze({
  PRIMARY: 'Primary',
  BASIC: 'Basic',
  OUTLINED: 'Outlined',
  GHOST: 'Ghost',
})

export const BUTTON_VARIATIONS = Object.freeze({
  FULL_WIDTH: 'FullWidth',
  AUTO_HEIGHT: 'AutoHeight',
  DEFAULT: 'Default',
})
