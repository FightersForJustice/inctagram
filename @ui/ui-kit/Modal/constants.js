export const CLOSE_MODAL_TEST_ID = 'closeModal'
