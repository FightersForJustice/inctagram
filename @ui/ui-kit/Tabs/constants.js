export const TAB_COLORS = Object.freeze({
  PRIMARY: 'Primary',
  GHOST: 'Ghost',
})
export const profileSettingsData = [
  { value: 'general', label: 'General information' },
  { value: 'Devices', label: 'Devices' },
  { value: 'AccountManagment', label: 'Account Managment' },
  { value: 'Payments', label: 'MyPayments' },
]
