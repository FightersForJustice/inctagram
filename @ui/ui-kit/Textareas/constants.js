export const TEXTAEREA_COLORS = Object.freeze({
  DEFAULT: 'Default',
  ACTIVE: 'Active',
  ERROR: 'Error',
})
