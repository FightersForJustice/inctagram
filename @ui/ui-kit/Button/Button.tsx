import React, { MouseEvent } from 'react'
import classNames from 'classnames'
import styles from './Button.module.scss'
import { BUTTON_COLORS } from './constants'

export type ButtonType = React.HTMLAttributes<HTMLButtonElement> & {
  text: string
  type?: 'button' | 'submit' | 'reset' | undefined
  autoHeight?: boolean
  color?: (typeof BUTTON_COLORS)[keyof typeof BUTTON_COLORS]
  disabled?: boolean
}

export const Button: React.FC<ButtonType> = (props) => {
  const { text = '', color = BUTTON_COLORS.PRIMARY, autoHeight = false, type = 'button' } = props
  const buttonClasses = classNames(styles.button, {
    [styles[`button${color}`]]: Boolean(color),
    [styles[`buttonAutoHeight`]]: Boolean(autoHeight),
  })

  return (
    <button className={buttonClasses} {...props} type={type}>
      {text}
    </button>
  )
}
