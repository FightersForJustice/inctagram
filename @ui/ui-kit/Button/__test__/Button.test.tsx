import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { Button } from '../Button'
import { BUTTON_COLORS } from '../constants'


describe('Button', () => {
  it('renders correctly', () => {
    const { getByText } = render(<Button text="Click me" />)
    const buttonElement = getByText('Click me')

    expect(buttonElement).toBeInTheDocument()
  })

  it('calls onClick handler when clicked', () => {
    const onClickMock = jest.fn()
    const { getByText } = render(<Button text="Click me" onClick={onClickMock} />)
    const buttonElement = getByText('Click me')

    fireEvent.click(buttonElement)

    expect(onClickMock).toHaveBeenCalledTimes(1)
  })

  it('applies custom color class', () => {
    const { container } = render(<Button text="Click me" color={BUTTON_COLORS.BASIC} />)
    const buttonElement = container.querySelector('button')

    expect(buttonElement).toHaveClass("button" + BUTTON_COLORS.BASIC)
  })

  it('is disabled when disabled prop is true', () => {
    const { container } = render(<Button text="Click me" disabled={true} />)
    const buttonElement = container.querySelector('button')

    expect(buttonElement).toBeDisabled()
  })
})
