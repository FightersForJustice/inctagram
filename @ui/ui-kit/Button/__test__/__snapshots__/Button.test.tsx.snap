// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render snapshot correctly 	 text:clickMe 	
            color: Basic 	
            disabled: false 	
             1`] = `
<DocumentFragment>
  <button
    class="button buttonBasic"
    type="button"
  >
    clickMe
  </button>
</DocumentFragment>
`;

exports[`render snapshot correctly 	 text:clickMe 	
            color: Basic 	
            disabled: true 	
             1`] = `
<DocumentFragment>
  <button
    class="button buttonBasic"
    disabled=""
    type="button"
  >
    clickMe
  </button>
</DocumentFragment>
`;

exports[`render snapshot correctly 	 text:clickMe 	
            color: Ghost 	
            disabled: false 	
             1`] = `
<DocumentFragment>
  <button
    class="button buttonGhost"
    type="button"
  >
    clickMe
  </button>
</DocumentFragment>
`;

exports[`render snapshot correctly 	 text:clickMe 	
            color: Ghost 	
            disabled: true 	
             1`] = `
<DocumentFragment>
  <button
    class="button buttonGhost"
    disabled=""
    type="button"
  >
    clickMe
  </button>
</DocumentFragment>
`;

exports[`render snapshot correctly 	 text:clickMe 	
            color: Outlined 	
            disabled: false 	
             1`] = `
<DocumentFragment>
  <button
    class="button buttonOutlined"
    type="button"
  >
    clickMe
  </button>
</DocumentFragment>
`;

exports[`render snapshot correctly 	 text:clickMe 	
            color: Outlined 	
            disabled: true 	
             1`] = `
<DocumentFragment>
  <button
    class="button buttonOutlined"
    disabled=""
    type="button"
  >
    clickMe
  </button>
</DocumentFragment>
`;

exports[`render snapshot correctly 	 text:clickMe 	
            color: Primary 	
            disabled: false 	
             1`] = `
<DocumentFragment>
  <button
    class="button buttonPrimary"
    type="button"
  >
    clickMe
  </button>
</DocumentFragment>
`;

exports[`render snapshot correctly 	 text:clickMe 	
            color: Primary 	
            disabled: true 	
             1`] = `
<DocumentFragment>
  <button
    class="button buttonPrimary"
    disabled=""
    type="button"
  >
    clickMe
  </button>
</DocumentFragment>
`;
